### PR TITLE
TracePluginProperty fixes

### PR DESCRIPTION
**Pull request by erlingrj**
When trying to compile a simple example with an external trace plugin I got linker errors. It seems that we need to link with `dl`. I am not exactly sure why it has worked before without specifying this though.
- [x] Fix problem where the LF_TRACE_PLUGIN Cmake variable is not initialized when using docker
- [x] Fix displaying of the trace-plugin target property in lfc (https://github.com/lf-lang/lingua-franca/issues/2273)

### CODE REVIEW

**Comment by lhstrh** [comment]
The fixmes in the description of this PR seem things that need to be addressed on the Lingua Franca end, are they not?

File: core/CMakeLists.txt [hunk 1]
```diff
@@ -62,6 +62,7 @@ if (DEFINED LF_TRACE)
     message(STATUS "linking trace plugin library ${LF_TRACE_PLUGIN}")
     target_link_libraries(reactor-c PUBLIC lf::trace-api)
     target_link_libraries(reactor-c PRIVATE "${LF_TRACE_PLUGIN}")
+    target_link_libraries(reactor-c PRIVATE dl)
```
Comment (lhstrh) [inline core/CMakeLists.txt:65]: :magic_wand:

**Review by lhstrh — APPROVED** [review]

File: core/CMakeLists.txt (hunk 1 above)
Comment (erlingrj) [reply core/CMakeLists.txt:65]: Added a brief commet

Comment (lhstrh) [reply core/CMakeLists.txt]: :+1:

**Review by lhstrh — APPROVED** [review]
Looks great! Thanks, @erlingrj.

**Comment by erlingrj** [comment]
I am going to double check why this was working in other projects.



ons. 8. mai 2024 kl. 22:13 skrev Marten Lohstroh ***@***.***>:

**Review by petervdonovan — APPROVED** [review]
LGTM

**Pull request merged**